pipeline-modules/deployment-service/gcp/pipeline-config.yaml

```diff
@@ -110,12 +110,17 @@ template: |
       export PATH="{{workspace}}/bin:$$PATH"
 
       echo "Preparing user environment variables..."
-      {% for env_key, env_value in pipeline_env %}
-      printf '{{env_key}}=%s\n' "$(printf '%s' "$${{env_key}}" | jq -aRs .)" >> {{pipeline_env_file}}
-      {% endfor %}
-      {% for env_key, env_value in pipeline_secret_env %}
-      printf '{{env_key}}=%s\n' "$(printf '%s' "$${{env_key}}" | jq -aRs .)" >> {{pipeline_env_file}}
-      {% endfor %}
+      exported_variable_names=({% for env_key in pipeline_env %} "{{env_key}}" {% endfor %}{% for env_key in pipeline_secret_env %} "{{env_key}}" {% endfor %})
```

Comment:
Isn't this still going to put all the variable names into the step? I guess it won't contain the values so that should shorten the overall length.

Reply:
Yes. It is however just names so I hope we won't hit it. Otherwise we could remove this but then we would be passing all envs. Unless we find a way to do this without printing the values.